error-budgets/SLOs-SLIs.md

```diff
@@ -0,0 +1,64 @@
+# Definitions of SLOs and SLIs for packit
+
+If you are not familiar with SLOs and SLIs, please check [README.md](./README.md) first.
+
+## How do others do it?
+
+The CKI team has a pretty comprehensive set of
+[SLOs](https://cki-project.org/docs/hacking/rfcs/cki-004-slo/#created-slos-based-on-the-feedback).
+Instead of taking the same route, let's start small and slowly add more and
+more objectives and indicators over time.
+
+Testing Farm team has [Service level
+expectations](https://docs.google.com/document/d/1iHK3v_tCK4w-F82ZvUEO3xHcNv78tkR31w3oF-zpi1Q/edit#heading=h.26gd46u8e15q)
+and a pretty slick
+[dashboard](http://grafana-latest-osci-infra.cloud.privileged.psi.redhat.com/d/VXwH27XMk/rhel-infrastructure-and-service-health?orgId=1).
+
+Cockpit also uses this framework and the team has [a nice dashboard as well](http://grafana-latest-osci-infra.cloud.privileged.psi.redhat.com/d/VgtdtiwGk/rhel-infrastructure-and-service-health?orgId=1).
+
+## Packit SLOs
+
+SLOs are based on [discussions with our stakeholders](users-expectations.md).
+
+Let's define job statuses first to get a clear understanding of these objectives:
+
+- successful run: a job finished and everything is awesome
+- failure: the job finished and wasn't successful: the PR needs to be fixed
+- error: infrastructure problems prevented the job to complete
+
+### 99% of builds finish with success/failure by 24 hours
```

Comment:
```suggestion
### 99% of builds finish with success or failure within 24 hours
```

... but this seem to somewhat contradict the practice to babysit copr build for about [8 hours](https://github.com/packit/packit-service/blob/ff58a490b6e78a57e6f5655b063ef3ccf4c6addf/packit_service/worker/tasks.py#L73). Maybe 8 hours is too little, but 24 hours seems too long, if I think about this in the context of CI. Worst case scenario, I would expect 12 hours. Anything above this seems to render any result irrelevant. But this is all speculation, probably would make sense to ask some prominent users, what do they expect. What is the maximum time for a build to finish, which is still useful for them and doesn't slow down their development too much.

Reply:
The problem here is that every project is different and builds take different times for ogr, cockpit, anaconda, kernel, systemd, atlas, gcc... One absolute number can't fit all. 

The indicator I'd like to see here would measure [moving average (MA)](https://en.wikipedia.org/wiki/Moving_average) and the object would require have consistent. Not sure how hard would it be to [measure MA](https://stackoverflow.com/questions/27489056/is-there-a-way-to-have-a-moving-average-in-grafana) with grafana.